src/pymoca/backends/casadi/api.py

```diff
@@ -366,19 +366,39 @@ def load_model(model_folder: str, model_name: str, compiler_options: Dict[str, s
 
             duration_dependencies = db['__delay_duration_dependent']
 
+            # Get rid of false dependency symbols not used in any delay
+            # durations. This significantly reduces the work the (slow)
+            # substitute() calls have to do later on.
```

Comment:
Some additional explanation as to why this is the case would be helpful.

Reply:
Why we get those false dependencies you mean, or why we want to get rid of them?

Reply:
Reading this code, I was left with the following question: Why does getting rid of the false dependencies have a significant impact on performance?